scaling_performance/host_practices.adoc

```diff
@@ -57,42 +57,106 @@ include writing a large amount of log data.
 configuration.
 
+|======================
+|OpenShift Version |etcd version|storage schema version
+|3.3 and earlier |2.x | v2
+|3.4 and 3.5 |3.x | v2
+|3.6 |3.x | v2 (upgrades)
+|3.6 |3.x | v3 (new installs)
+|======================
+
+etcd 3.x brings with it important scalability and performance improvements that reduce CPU, memory, network and disk requirements for any size cluster.  etcd 3.x also implements a backwards compatible storage API that facilitates a two-step migration of the on-disk etcd database.  For migration purposes, the storage mode used by etcd 3.x in OpenShift 3.5 remained in v2 mode.  As of OpenShift 3.6, new installs will use storage mode v3.  Upgrades from previous versions of OpenShift will NOT automatically migrate data from v2 to v3.  This must be done using supplied playbooks following the documented process.
+
+Version 3 of etcd implements a backwards compatible storage API that facilitates a two-step migration of the on-disk etcd database.  For migration purposes, the storage mode used by etcd 3.x in OpenShift 3.5 remained in v2 mode.  As of OpenShift 3.6, new installs will use storage mode v3.  In order to provide customers time to prepare for migrating the etcd schema from v2 to v3 (and associated downtime and verification), OpenShift 3.6 does not enforce this upgrade.  However, based on extensive test results Red Hat strongly recommends migrating existing OpenShift clusters to etcd 3.x storage mode v3.  This is particularly relevant in larger clusters, or in scenarios where SSD storage is not available.
+
+[IMPORTANT]
+====
+etcd schema migration will be required by future OpenShift upgrades.
+====
+
+In addition to changing the storage mode for new installs to v3, OpenShift 3.6 also begins enforcing "quorum reads" for all OpenShift types.  This is done to ensure that queries against etcd do not return stale data.  In single-node etcd clusters, stale data is not a concern.  In highly available etcd deployments typically found in production clusters, quorum reads ensure valid query results.  A quorum read is "linearizable" in database terms - every client sees the latest updated state of the cluster, and all clients see the same sequence of reads and writes.  Please see the etcd 3.1 https://coreos.com/blog/etcd-3-1-announcement.html[announcement] for more information on performance improvements.
+
+It is important to note that OpenShift uses etcd for storing additional information beyond what Kubernetes itself requires.  For example, OpenShift stores information about images, builds (amongst others) in etcd as is required by features that OpenShift adds on top of Kubernetes.  Ultimately what this means is that guidance around performance and sizing for etcd hosts will differ from Kubernetes and other recommendations in salient ways.  Red Hat tests etcd scalability and performance with the OpenShift use-case and parameters in mind to generate the most accurate recommendations.
```

Comment:
s/OpenShift/{product-title}
s/builds (amongst others)/ builds, and other components 
etcd, as is required
Ultimately, this means that